admin/listPurchases/listView.html

```diff
@@ -77,13 +88,142 @@
         }
-
         return false;
     });
 });
 
+const amountOfShownSubpages = 5;
+const entriesPerSubpage = 20;
+let amountOfSubpages;
+
+function generateSubPageNavigation(){
+    if(orderSearchBar.value){
+        amountOfSubpages = Math.ceil(entries.filter(currentOption.filter).filter(searchBarFilter).length / entriesPerSubpage);
+    }else{
+        amountOfSubpages = Math.ceil(entries.filter(currentOption.filter).length / entriesPerSubpage);
+    }
+
+    if(currentPageIndex >= amountOfSubpages){
+        currentPageIndex = amountOfSubpages - 1;
+    } 
+
+    let startIndexToRender = 0;
+    let subpageNavigation = [];
+
+    const extraNavNeeded = amountOfSubpages > amountOfShownSubpages;
+
+    if(extraNavNeeded){
+        subpageNavigation.push(`<a href="#" class="option subpage-selector subpage-selector-to-start">&lt&lt</a>`);
+        subpageNavigation.push(`<a href="#" class="option subpage-selector subpage-selector-backwards">&lt</a>`);
+
+        startIndexToRender = currentPageIndex;
+        if(startIndexToRender + Math.floor(amountOfShownSubpages/2) >= amountOfSubpages){
+            startIndexToRender = amountOfSubpages - amountOfShownSubpages;
+        }else if(startIndexToRender - Math.floor(amountOfShownSubpages/2) > 0){
+            startIndexToRender -= Math.floor(amountOfShownSubpages/2);
+        }else{
+            startIndexToRender = 0;
+        }
+    }
+    
+    for(let i = startIndexToRender; i < startIndexToRender + amountOfShownSubpages && i < amountOfSubpages; i++){
+        subpageNavigation.push(`<a href="#" class="option subpage-selector subpage-selector-index-${i}">${i+1}</a>`);
+    }
+
+    if(extraNavNeeded){
+        subpageNavigation.push(`<a href="#" class="option subpage-selector subpage-selector-forward">&gt</a>`);
+        subpageNavigation.push(`<a href="#" class="option subpage-selector subpage-selector-to-end">&gt&gt</a>`);
+    }
+
+    nav.innerHTML = subpageNavigation.join("");
+
+    if(extraNavNeeded){
+        document.querySelector(`.subpage-selector-backwards`).addEventListener("click", function (e){
+            e.preventDefault();
+            if(currentPageIndex <= 0) return;
+            currentPageIndex--;
+            updateContent();
+        });
+        document.querySelector(`.subpage-selector-forward`).addEventListener("click", function (e){
+            e.preventDefault();
+            if(currentPageIndex >= amountOfSubpages - 1) return;
+            currentPageIndex++;
+            updateContent();
+        });
+        document.querySelector(`.subpage-selector-to-start`).addEventListener("click", function (e){
+            e.preventDefault();
+            currentPageIndex = 0;
+            updateContent();
+        });
+        document.querySelector(`.subpage-selector-to-end`).addEventListener("click", function (e){
+            e.preventDefault();
+            currentPageIndex = amountOfSubpages - 1;
+            updateContent();
+        });
```

Comment:
All of these can be generalized to something like this (where`registerNavigationButton` is 8 lines of code including the wrapping function definition lines):

```js
registerNavigationButton(`.subpage-selector-backwards`, (n) => n - 1);
registerNavigationButton(`.subpage-selector-forward`, (n) => n + 1);
registerNavigationButton(`.subpage-selector-to-start`, (n) => 0);
registerNavigationButton(`.subpage-selector-to-end`, (n) => amountOfSubpages - 1);
```

Remember to do the check for whether the page number is within range, still.